charts/opentelemetry-operator/UPGRADING.md

```diff
@@ -1,5 +1,9 @@
 # Upgrade guidelines
 
+## 0.21 to 0.22.0
+Kubernetes resource names will now use `{{opentelemetry-operator.fullName}}` as the default value which will change the name of many resources.
+Some CI/CD tools might create duplicate resources when upgrading from an older version because of this change.
```

Comment:
Can you please also provide guidance on how to keep the existing names for an upgrade? Should be possible with `nameOverride`, right?

Reply:
@dmitryax because so many of the objects had hard-coded postfixes I don't think we can use `nameOverride`.  The mutating webhooks are an example.  Same with clusterrole and clusterrolebinding.

Reply:
It should be fine to rename `clusterrole` and `clusterrolebinding` during the upgrade. I'm mostly worried about deployments. Just wanted to make sure we provide a non-disruptive upgrade experience

Reply:
In that case yes, we can give recommendations on how to keep the original deployment name with nameOverride.  @sfenman can you make that change to UPGRADING.md?

Reply:
And please test that recommendation. The goal is to have a non-blocking upgrade experience. AFAIR, you cannot change a deployment name during the upgrade

Reply:
Yes, I will test it. AFAIK deployment will be replaced during the `helm upgrade` if it has a new name. Also, I think users should use `fullnameOverride` instead of `nameOverride` to get the same deployment name.